lib/http2/connection.c

```diff
@@ -617,10 +617,20 @@ static int handle_incoming_request(h2o_http2_conn_t *conn, h2o_http2_stream_t *s
     assert(stream->state == H2O_HTTP2_STREAM_STATE_RECV_HEADERS);
 
     if ((ret = h2o_hpack_parse_request(&stream->req.pool, h2o_hpack_decode_header, &conn->_input_header_table,
-                                       &stream->req.input.method, &stream->req.input.scheme, &stream->req.input.authority,
-                                       &stream->req.input.path, &stream->req.upgrade, &stream->req.headers, &header_exists_map,
-                                       &stream->req.content_length, &expect, &stream->cache_digests, NULL, src, len, err_desc)) !=
-        0) {
+                                       (h2o_hpack_request_t){
+                                           .method = &stream->req.input.method,
+                                           .scheme = &stream->req.input.scheme,
+                                           .authority = &stream->req.input.authority,
+                                           .path = &stream->req.input.path,
+                                           .protocol = &stream->req.upgrade,
+                                           .headers = &stream->req.headers,
+                                           .pseudo_header_exists_map = &header_exists_map,
+                                           .content_length = &stream->req.content_length,
+                                           .expect = &expect,
+                                           .digests = &stream->cache_digests,
+                                           .datagram_flow_id = NULL,
+                                       },
```

Comment:
I get the fuss, but I'm not sure if I like the direction of this PR.

As can be seen in these lines of changes, essentially this PR is changing making all arguments optional. When an argument is missing, they will be silently default-initialized (i.e., set to either 0 or NULL), rather than failing at compile time.

Is that the behavior we want? Assuming that we change the arguments rarely, my preference goes to detecting those changes as compile errors so that we can make sure we are doing the correct thing.

WDYT?

Reply:
I wonder if we could use https://gcc.gnu.org/onlinedocs/gcc-13.2.0/gcc/Warning-Options.html#index-Wmissing-field-initializers to catch this kind of errors?